pulpcore/cli/rpm/comps.py

```diff
@@ -0,0 +1,65 @@
+import gettext
+from typing import IO, Optional
+
+import click
+
+from pulpcore.cli.common.context import (
+    EntityFieldDefinition,
+    PluginRequirement,
+    PulpContext,
+    PulpEntityContext,
+    pass_entity_context,
+    pass_pulp_context,
+)
+from pulpcore.cli.common.generic import resource_option
+from pulpcore.cli.rpm.context import PulpRpmCompsXmlContext, PulpRpmRepositoryContext
+
+_ = gettext.gettext
+
+repository_option = resource_option(
+    "--repository",
+    default_plugin="rpm",
+    default_type="rpm",
+    context_table={"rpm:rpm": PulpRpmRepositoryContext},
+    href_pattern=PulpRpmRepositoryContext.HREF_PATTERN,
+    help=_("Repository to associate the comps-units to, takes <name> or href."),
+)
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "repo_type",
+    type=click.Choice(["rpm"], case_sensitive=False),
+    default="rpm",
+)
+@pass_pulp_context
+@click.pass_context
+def comps(ctx: click.Context, pulp_ctx: PulpContext, repo_type: str) -> None:
+    if repo_type == "rpm":
+        pulp_ctx.needs_plugin(PluginRequirement("rpm", min="3.17.0dev"))
+        ctx.obj = PulpRpmCompsXmlContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+@comps.command()
+@click.option("--file", type=click.File("rb"), required=True)
+@repository_option
+@click.option("--replace", type=bool, default=False)
+@pass_entity_context
+@pass_pulp_context
+def upload(
+    pulp_ctx: PulpContext,
+    entity_ctx: PulpRpmCompsXmlContext,
+    file: IO[bytes],
+    repository: Optional[EntityFieldDefinition],
+    replace: Optional[bool],
+) -> None:
+    """Create an rpm package content unit by uploading a file"""
```

Comment:
This is probably wrong. Keep in mind that this docstring is user facing.

Reply:
Ooo, nice catch - copypasta strikes again!